server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/fileformats/node/FixedLengthStarTreeNode.java

```diff
@@ -233,68 +241,155 @@ private static FixedLengthStarTreeNode matchStarTreeNodeTypeOrNull(FixedLengthSt
         }
     }
 
+    @Override
+    public Iterator<FixedLengthStarTreeNode> getChildrenIterator() throws IOException {
+        return new Iterator<>() {
+            private int currentChildId = firstChildId;
+            private final int lastChildId = getInt(LAST_CHILD_ID_OFFSET);
+
+            @Override
+            public boolean hasNext() {
+                return currentChildId <= lastChildId;
+            }
+
+            @Override
+            public FixedLengthStarTreeNode next() {
+                try {
+                    return new FixedLengthStarTreeNode(in, currentChildId++);
+                } catch (IOException | RuntimeException e) {
+                    throw new IllegalStateException(e);
+                }
+            }
+
+            @Override
+            public void remove() {
+                throw new UnsupportedOperationException();
+            }
+        };
+    }
+
     /**
-     * Performs a binary search to find a child node with the given dimension value.
+     * Finds and returns all children with dimension values between the given start and end values.
      *
-     * @param dimensionValue The dimension value to search for
-     * @return The child node if found, null otherwise
+     * @param startDimensionValue The start of the range (inclusive)
+     * @param endDimensionValue   The end of the range (inclusive)
+     * @return A list of child nodes whose dimension values lie between the specified range
      * @throws IOException If there's an error reading from the input
      */
-    private FixedLengthStarTreeNode binarySearchChild(long dimensionValue) throws IOException {
+    public Iterator<FixedLengthStarTreeNode> range(long startDimensionValue, long endDimensionValue) throws IOException {
```

Comment:
As discussed offline, we need to support list of ranges, with base case being just one range. Lets probably see if we can have a Range class similar to the one in RangeAggregator.